ci/pipeline-staging.yml

```diff
@@ -0,0 +1,170 @@
+      - across:
+        - var: build-task
+          values: ((.:build-tasks))
+        do:
+        - task: build
+          privileged: true
+          config:
+            platform: linux
+            image_resource:
+              type: registry-image
+              source:
+                repository: concourse/oci-build-task
+            inputs:
+            - name: src
+              path: .
+            outputs:
+            - name: image
+            run:
+              path: build
+            params:
+              BUILD_ARG_TASK_FOLDER: tasks/((.:build-task))
+              BUILD_ARGS_FILE: tasks/((.:build-task))/.env
+              BUILDKIT_SECRETTEXT_UA_TOKEN: ((ua-token))
+              DOCKERFILE: Dockerfile
+        - task: tag-list
+          file: src/ci/partials/tag-list.yml
+          image: cf-image
+          params:
+            TAG: ((.:build-task))-((deploy-env))
+        - put: image-repository-((deploy-env))
+          params:
+            image: image/image.tar # this might need output_mapping
+            additional_tags: src/tag-list.txt
+        - task: deploy
+          config:
+            <<: *cf-image
+            inputs:
+              - name: src
+              - name: image-repository-((deploy-env))
+            run:
+              dir: src
+              path: ci/tasks/deploy.sh
+          params:
+            <<: *env-cf
+            CF_APP_NAME: pages-((.:build-task))-task-((deploy-env))
+            CF_MANIFEST: .cloudgov/manifest.yml
+            CF_VARS_FILE: .cloudgov/vars/pages-((deploy-env)).yml
+            IMAGE_REPOSITORY: ../image-repository-((deploy-env))/repository
+            IMAGE_TAG: ((.:build-task))-((deploy-env))
+            CF_DOCKER_USERNAME: ((ecr-aws-key))
+            CF_DOCKER_PASSWORD: ((ecr-aws-secret))
+          on_failure:
+            put: slack
+            params:
+              text: |
+                :x: FAILED: pages-cf-build-tasks: ((.:build-task)) deployment on ((deploy-env))
+                <$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|View build details>
+              channel: ((slack-channel))
+              username: ((slack-username))
+              icon_url: ((slack-icon-url))
+          on_success:
+            put: slack
+            params:
+              text: |
+                :white_check_mark: SUCCESS: Successfully deployed pages-cf-build-tasks: ((.:build-task)) on ((deploy-env))
+                <$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|View build details>
+              channel: ((slack-channel))
+              username: ((slack-username))
+              icon_url: ((slack-icon-url))
+
+
+############################
+#  RESOURCES
+
+resources:
+
+  - name: src-((deploy-env))
+    type: git
+    icon: github
+    source:
+      uri: cloud-gov/pages-cf-build-tasks
+      branch: main
```

Comment:
In other places we've been using the `harden-concourse-task` image, defining it as a resource_type with
```
  - name: pull-request
    type: registry-image
    source:
      aws_access_key_id: ((ecr_aws_key))
      aws_secret_access_key: ((ecr_aws_secret))
      repository: github-pr-resource
      aws_region: us-gov-west-1
      tag: latest
```
down below and then as a resource here with
```
    type: pull-request
    check_every: 1m
    source:
      repository: ((proxy-repository-path))
      access_token: ((gh-access-token))
      base_branch: ((git-branch))
      disable_forks: true
      ignore_drafts: false
```
There may be a reason you're not doing that, but I wanted to to make sure.

Reply:
That's not a pull request resource but a git resource. We'll want to add the hardened git resource type.

```yml
  - name: git
    type: registry-image
    source:
      aws_access_key_id: ((ecr_aws_key))
      aws_secret_access_key: ((ecr_aws_secret))
      repository: git-resource
      aws_region: us-gov-west-1
      tag: latest
```